### ISSUE TITLE
Error when running Benzene along with the production of *.aoints.* and *.gridpts.* files

### ISSUE THREAD

**Issue by hawiki** [nwchemgit/nwchem#483]
**Describe the bug**
When I run C6H6 (benzene) simulation using Hartree-Fock, I got this error

    Integral file          = ./C6H6.aoints.0
    Record size in doubles =    65536    No. of integs per rec  =    32766
    Max. records in memory =        0    Max. records in file   = ********
    No. of bits per label  =       16    No. of bits per value  =       64

     eaf_write: rc ne bytes -1999 bytes 524288
     IO offset    35127296.0000000     
     IO error message >Write Failed
     ------------------------------------------------------------------------
      int2e_packed_buf_write: write failed                   0
     ------------------------------------------------------------------------
     ------------------------------------------------------------------------
      current input line : 
      32: task dft
     ------------------------------------------------------------------------
     ------------------------------------------------------------------------
     This error has not yet been assigned to a category
     ------------------------------------------------------------------------

First of all, not sure if it's a bug or not but as I recall it happens after I accidentally make changes to a source file (not a big change, it accidental) but I couldn't remember which source file it was because my editor closed suddenly. Afterwards, whenever I run a HF job on multiple cores I have *aoints.0, *aoints.1, ... *aoints.n files in the job's directory where n-1 is the number of cores. When I run DFT job I would also have *.gridpts.0, ... , *gridpts.n. I don't feel like I have these files in the job's folder before that accidental change. Strangely, when I do calculation with smaller molecule such as CH3F, these files got produced as well but I don't have the above error.

Could someone please point out, based on the error message above, which source file should I look for the issue?

**Describe settings used**
I am on a computer cluster running on Red Hat. The version is NWChem 6.8.

**Comment by hawiki** [comment]
UPDATE: It turns out that I am running out of storage space because those integral and grid files are of big size. One integral file for benzene calculation using cc-pvtz basis can be as large as 5-6 GB. So, if I use say 8 cores, they take up ~48 GB. But the problem is clear now, it's because these files are printed. I thought that the 2e integrals are stored in memory as long as it fits, instead of being printed to hard disk. But I am sure I don't have any directives in my input file that instruct NWChem to print these files out.

**Comment by jeffhammond** [comment]
You should use direct SCF. Add "direct" to scf input block. Or dft if you're doing that task. I'm surprised it isn't the default.

**Comment by hawiki** [comment]
Thanks for the reply.
I use NWChem 6.8.
Quoting the documentation on the direct/semidirect directive under Hartree-Fock section (which is still the same between v6.6 that I have and the one online, so I guess v6.8 doc should be the same as well):

_The default behavior of the SCF module is_
1. _If enough memory is available, the integrals are computed once and are cached in memory._
2. _If there is not enough memory to store all the integrals at once, then 95% of the available disk space in the scratch directory (see File directories) is assumed to be available for this purpose, and as many integrals as possible are cached on disk (with no memory being used for caching). Some attempt is made to store the most expensive integrals in the cache._
3. _If there is not enough room in memory or on disk for all the integrals, then the ones that are not cached are recomputed in a semidirect fashion._

According to this, therefore even if I don't have enough disk space or memory, the code should recompute the integrals every iteration instead of encountering an error and terminating. So, manually using `direct` directive should not be necessary.
And what about the *.gridpts.* file. Is it by default printed and not deleted at the end of the simulation?

**Comment by jeffhammond** [comment]
Just add direct or semidirect to make it clear. That's the safest choice.

Temporary files should get deleted on successful termination. They probably won't be on a crash. Best to put a cleanup command in your job script.

**Comment by hawiki** [comment]
Yes in the meantime I have always put `rm -f *gridpts.*` at the end of my job script.
I tried using `semidirect filesize x`. Putting this in input, I expect that the *.aoints.* file will still get printed but their individual size is capped at a byte size corresponding to the word length `x`. But turns out that now the program prints *.gridpts.* instead of *.aoints.* files. Looks like in the condition I requested through the `semidirect` line above, the program uses numerical integration to obtain the 2e integrals? I don't know. But it does resolve the issue in that the scf iteration runs till convergence. The behavior pertaining to the `direct | semidirect` directive is a bit confusing though. 
Anyway, is there an easy way to estimate the 2e integrals disk size in bytes instead of 64-bit words. Because when I encounter the error above, I don't always get the `#quartets=` line in the output which contains info about the total no. of integrals. And in this case I have no clue what integer I should put for the `semidirect filesize` directive. On the other hand, since I know how much disk space I have in bytes, I can easily tell how much max. byte each integral files should have for each process without the need to know the no. of integrals.
